client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/JooqBackfill.kt

```diff
@@ -0,0 +1,147 @@
+package app.cash.backfila.client.misk.jooq
+
+import app.cash.backfila.client.misk.Backfill
+import app.cash.backfila.client.misk.BackfillConfig
+import app.cash.backfila.protos.clientservice.KeyRange
+import com.google.common.base.Preconditions
+import com.squareup.backfila.client.base.jooq.ByteStringSerializer
+import com.squareup.backfila.client.base.jooq.CompoundKeyComparer
+import com.squareup.backfila.client.base.jooq.CompoundKeyComparisonOperator
+import okio.ByteString
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.Field
+import org.jooq.Record
+import org.jooq.SortField
+import org.jooq.TableLike
+
+/**
+ * Implement this interface in your project to run a backfill with Jooq
+ *
+ * @param <K> the type of key being backfilled over
+ * @param <Param> backfill parameters
+*/
+interface JooqBackfill<K, Param: Any>: Backfill {
+  /**
+   * A map of jooq transacters, indexed by shard database name, used to interact with the
+   * database(s). For the unsharded case, this will will be a map of one entry.
```

Comment:
With Jooq sharding do you ever have to be worried about resharding? Will your code fail correctly? Or is it always safe?

Reply:
That I think depends on the implementation of the JooqTransacter passed in. I havent really tested on a sharded db as I didn't have the time. Our services dont have a sharded db yet. I might have to come back to this when I have sometime.